### PR TITLE
fuse -> ROS 2 fuse_models: Linting

### PR DESCRIPTION
**Pull request by methylDragon**
See: #276
Depends on: #314

Linting
Migrates .h to .hpp
Pinging @svwilliams for visibility.

Note: There's an open question about what to do about test_unicycle_2d.cpp and test_sensor_proc.cpp's proprietary copyright licenses (to fix ament_copyright)

### CODE REVIEW

**Comment by methylDragon** [comment]
Note @svwilliams : Two files had to get excluded from linting: https://github.com/locusrobotics/fuse/pull/315/commits/e62784bcced763575354e6fe2f29b2f29e196371

This is because they featured copyright docstrings with proprietary copyrights.

https://github.com/methylDragon/fuse/blob/rolling-models-linting/fuse_models/test/test_sensor_proc.cpp
```
/***************************************************************************
 * Copyright (C) 2019 Clearpath Robotics. All rights reserved.
 * Unauthorized copying of this file, via any medium, is strictly prohibited
 * Proprietary and confidential
 ***************************************************************************/
```

https://github.com/methylDragon/fuse/blob/rolling-models-linting/fuse_models/test/test_unicycle_2d.cpp
```
/***************************************************************************
 * Copyright (C) 2017 Locus Robotics. All rights reserved.
 * Unauthorized copying of this file, via any medium, is strictly prohibited
 * Proprietary and confidential
 ***************************************************************************/
```

File: fuse_models/include/fuse_models/imu_2d.hpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+/**
+ * @brief An adapter-type sensor that produces orientation (relative or absolute), angular velocity,
+ *        and linear acceleration constraints from IMU sensor data published by another node
+ *
+ * This sensor subscribes to a sensor_msgs::msg::Imu topic and:
+ * 1. Creates relative or absolute orientation and constraints. If the \p differential parameter is
+ * set to false (the default), the orientation measurement will be treated as an absolute
+ * constraint. If it is set to true, consecutive measurements will be used to generate relative
+ * orientation constraints. 2. Creates 2D velocity variables and constraints.
```
Comment (sloretz) [inline fuse_models/include/fuse_models/imu_2d.hpp]: nit, newline to keep list structure.

File: fuse_models/include/fuse_models/odometry_2d.hpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+{
+
+/**
+ * @brief An adapter-type sensor that produces pose (relative or absolute) and velocity constraints
+ *        from sensor data published by another node
+ *
+ * This sensor subscribes to a nav_msgs::msg::Odometry topic and:
+ * 1. Creates relative or absolute pose variables and constraints. If the \p differential parameter
+ * is set to false (the default), the  measurement will be treated as an absolute constraint. If it
+ * is set to true, consecutive measurements will be used to generate relative pose constraints. 2.
```
Comment (sloretz) [inline fuse_models/include/fuse_models/odometry_2d.hpp]: same comment about new line to keep list structure.

File: fuse_models/include/fuse_models/unicycle_2d_state_cost_function.hpp [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+  /**
+   * @brief Evaluate the cost function. Used by the Ceres optimization engine.
+   *
+   * @param[in] parameters - Parameter blocks: 0 : position1 - First position (array with x at index
+   *                         0, y at index 1) 1 : yaw1 - First yaw 2 : vel_linear1 - First linear
+   *                         velocity (array with x at index 0, y at index 1) 3 : vel_yaw1 - First
+   *                         yaw velocity 4 : acc_linear1 - First linear acceleration (array with x
+   *                         at index 0, y at index 1) 5 : position2 - Second position (array with x
+   *                         at index 0, y at index 1) 6 : yaw2 - Second yaw 7 : vel_linear2 -
```
Comment (sloretz) [inline fuse_models/include/fuse_models/unicycle_2d_state_cost_function.hpp]: Newlines to preserve list structure here

File: fuse_models/include/fuse_models/unicycle_2d_state_cost_function.hpp [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+      // It might be possible to simplify the code below implementing something like this but using
+      // compile-time template recursion.
+      //
+      // // state1: (position1, yaw1, vel_linear1, vel_yaw1, acc_linear1)
+      // for (size_t i = 0; i < 5; ++i)
+      // {
+      //   if (jacobians[i])
+      //   {
+      // Eigen::Map<fuse_core::Matrix<double, 8, ParameterDims::GetDim(i)>> jacobian(jacobians[i]);
+      // jacobian.applyOnTheLeft(-A_); } }
```
Comment (sloretz) [inline fuse_models/include/fuse_models/unicycle_2d_state_cost_function.hpp]: Newlines to preserve code example here

**Review by sloretz — CHANGES_REQUESTED** [review]
All tests in fuse_publishers and fuse_models pass for me locally. When these comments are addressed LGTM

**Pull request merged**